### ISSUE TITLE
Odd hunger states

### ISSUE THREAD

**Issue by ZaricaDemoni** [CleverRaven/Cataclysm-DDA#41889]
# Describe the bug
Eating anything, or being anywhere below Satiated hunger level, means very hungry or nothing.

# Steps To Reproduce
Eat anything when having a blank hunger level
Become very hungry

# Expected behavior
Not being very hungry despite having eaten food.

# Screenshots
![image](https://user-images.githubusercontent.com/32645137/86553836-b113ee00-bf00-11ea-8046-dbd56f71c177.png)

# Versions and configuration

 - OS: Windows 8.1
 - Game Version: 0.E-3769-g1f55aa9
 - Graphics version: Tiles
 - Ingame language: English
 - Mods loaded: "dda", "no_npc_food", "blazeindustries", "crt_expansion", "magiclysm", "DinoMod", alt_map_key", "MMA", "cbm_slots", "fuji_mpp", "speedydex", "stats_through_kills", "StatsThroughSkills"

# Additional context
As seen in image I do have food on my stomach and in my gut, yet, I'm still considered very hungry. Only time Its not set to very hungry appears to be when Satiated, Full or Engorged. Watching it while slowly eating things It has done the following state change: Very Hungry - Nothing - Very Hungry - Satiated - Full. which is kind of unusual.

**Comment by ZaricaDemoni** [comment]
Just noticed the in-progress build mentions fixing nutrients/stomach. This may be closed when its finished and checked.

**Comment by anothersimulacrum** [comment]
It won't be.

**Comment by UmbralReaper** [comment]
Here is a chart of the conditions the game checks to determine displayed hunger :
```
if stomach is > ~83% full, then hunger = engorged
if stomach is > 55% full, then hunger = full
if stomach is > 37.5% full & ate < 3 hours ago, then hunger = satisfied
if you ate > 15 minutes ago & either ate within the last 3 hours or have something in your stomach, then hunger = blank
if you are overweight then hunger = hungry
else hunger = very hungry
```
And here is a chart of the conditions to determine displayed hunger for underweight characters :
```
if stomach is > 100% full, then hunger = engorged
if stomach is > 75% full, then hunger = full
if stomach is > 50% full & ate < 15 minutes ago, then hunger = satisfied
if ate within 15 minutes, then hunger = hungry
if ate within 3 hours, then hunger = very hungry
else hunger = famished
```
The reason for your constantly shifting state of hunger is because your character is underweight. Following the chart above, it can be seen that even if you are 50% full, as long as you haven't eaten in the last 15 minutes, you are very hungry.

In other words, this is not a bug, but the game working as intended.

**Comment by ZaricaDemoni** [comment]
That would open up a whole new bug on its own for another issue if what you say is correct. Give the screenshot i just took (only 3kcal difference) is the same state my character was in, during the previous screenshot, and when i finally did hit 39kcal stored area. however, being able to make a new character at the time of opening this issue, spawning with food and eating it should never make you immediately hungry which was the intent of the original post. but now I'm curious if the player stat weight label if broken.

![image](https://user-images.githubusercontent.com/32645137/87231254-74b21900-c36a-11ea-8a34-751b544db5e8.png)

**Comment by UmbralReaper** [comment]
Incorrectly displayed weight certainly does warrant a new issue (if you can reproduce it). 

> however, being able to make a new character at the time of opening this issue, spawning with food and eating it should never make you immediately hungry which was the intent of the original post

This is probably occurring because the game checks stomach capacity and updates when you eat.

> Eat anything when having a blank hunger level.

Can you provide a screenshot with the same debug menu message (side note: how do you get that to display?) and a blank hunger level? Underweight characters should never have a blank hunger level.

**Comment by ZaricaDemoni** [comment]
freshly spawned character, hobo which starts with can of beans, I've done nothing but run the debug menu to check game state.

![image](https://user-images.githubusercontent.com/32645137/87244860-4d545e00-c3f5-11ea-86c0-3f932c377015.png)

I've eaten the beans in his invantory

![image](https://user-images.githubusercontent.com/32645137/87244874-7543c180-c3f5-11ea-9420-0d1ce33041e4.png)

(for some reason he spawned slightly overweight)

second test, again went with hobo, but made him underweight to start (though you can only choose XS, meaning im near death level underweight). Here it can be seen working properly that im labled as starving to start with. 

![image](https://user-images.githubusercontent.com/32645137/87244940-fef38f00-c3f5-11ea-865a-feda9790dfa4.png)


and this is after eating the beans, drinking whiskey and gin. - Eating 1 thing of beans made it go from starving, immediately to "hungry" and didn't change from there until satisfied. which is where I'm currently sitting.

![image](https://user-images.githubusercontent.com/32645137/87244968-4d089280-c3f6-11ea-9b7e-fd80d6557520.png)

as for the last question, I added a hotkey to debug menu to help report bugs I find. Just go to debug, info, game state. press enter a few times and it will post stomach contents in the message bar.

**Comment by UmbralReaper** [comment]
Based upon that, everything seems to be working as intended (the last character actually had a BMI of 25). 

You are right though, eating shouldn't make you hungrier, so I'll take another look at the code.

**Comment by Alex-Folts** [comment]
There is really something not right with food, eat 5 pieces of meat each 401 cal and after 3 hours already very hungry, about 12-15 pieces of meat(its over 2 kg) or 100 meat scraps(i checked: 60 not enough, you already very hungry and its only 16-17 oclock, need 80-100 to make a day) need per day to not lose weight, If less, even if you eat something, 1-2 days and already underweight. Even when sleep, after 3 hours become very hungry.
And difference when you work or you read book whole day is minor, it feels like just idle calory consumption is very high.

**Comment by UmbralReaper** [comment]
Can you provide the calories used from the tracker added in #41997? If expenditure is really that high, it might be worth creating a new issue based upon calorie expenditure.

**Comment by Alex-Folts** [comment]
I'll try. I didn't know about tracker.

**Comment by Alex-Folts** [comment]
Here what i have now in tracker, but i cant say what i did those days so it is not tells much.
![Scr1133](https://user-images.githubusercontent.com/5638906/87556482-61cb7b80-c6bf-11ea-903f-eb97f5b9cc85.jpg)

What i wanna do, is use my last 50 protein rations(no meat left already) and test few days, where i will do nothing all day only eat and skip time, will read whole day and eat, and will walk all day with full backpack of stuff and also eat. Such kind of tests and see the results.

**Comment by anothersimulacrum** [comment]
Carry weight has no impact on calories spent.

**Comment by Alex-Folts** [comment]
> Also, do you have the fast metabolism, or any metabolic traits?
No traits, no mutations, no bionics.

> Carry weight has no impact on calories spent.
I think carried weight(including character's overweight) should scale calories spend when you moving. More mass to move requires more energy.
What about craft/building/digging, it has impact?

**Comment by anothersimulacrum** [comment]
It definitely should, it just doesn't.
Yes, different activities will have different impacts - digging is one of the more intensive ones (along with running).

**Comment by ZaricaDemoni** [comment]
New info regarding the original issue. eating when in blank state at BMI 24 causes very hungry status, but BMI 25 causes standard hungry. noticed this since i was fluctuating at that point. this is a very confusing issue.

**Comment by Alex-Folts** [comment]
So i tested:

- do nothing whole day, satay whole day in bed and only eat as much as it can every time character become very hungry(every 3 hours)
- whole day digging and eat as much as he can.
- whole day waliking and it when very hungry
- spend whole day making anvil, eat a lot too.

all this time character eat protein rations, i spawn 100, he ate 77 in this period, eat roughly every 3 hours from state "very hungry" to state "full"(yellow after which you cant eat anymore). Sleep 9 hours a day.

Here is numbers:
![Scr1136](https://user-images.githubusercontent.com/5638906/87590407-660f8d00-c6ef-11ea-9dbd-4475ad1d6e96.jpg)

From what i see it is very difficult to keep weight, even if you have infinite food, idk how to gain weight in this conditions... something really wrong.

**Comment by anothersimulacrum** [comment]
A whole day is 24 hours, or what? Do you sleep for 8 hours or so of the day, or what?

**Comment by Alex-Folts** [comment]
day 24 hours, sleep 9. While awake did said tasks, ~15 hours.

**Comment by ZaricaDemoni** [comment]
@Alex-Folts mind taking a screenshot of the message log when you run the debug? I'm curious what your BMI is, as i was dealing with similar at one point. (i was still considered normal weight, but had to eat way to much, I took an anti-parasitic but it didnt kill anything even-though it seemed to work. was around 40000 cal - 43000 cal)

**Comment by Alex-Folts** [comment]
> @Alex-Folts mind taking a screenshot of the message log when you run the debug?

This?
![Scr1138](https://user-images.githubusercontent.com/5638906/87702454-1c33af00-c7a2-11ea-8cc9-494af8b3941e.jpg)

**Comment by ZaricaDemoni** [comment]
if this is a character you care about, id suggest finding a lot of maple candies and potentially checking for a parasite. you are definatly under weight which is why it wants you to eat so much.

**Comment by Alex-Folts** [comment]
> you are definatly under weight

But i cant get weight.

here calory stats:
> ![Scr1133](https://user-images.githubusercontent.com/5638906/87556482-61cb7b80-c6bf-11ea-903f-eb97f5b9cc85.jpg)
each day deficite, dispite character eat normal food like meat, and now i run out of meat, and at this rate any other less energy dense food will make only worse.

I wonder how in such conditions play when you living of land/farming/hunting or fishing, it is almost seems like character would spend more energy to get food than will refil with that food...

**Comment by ZaricaDemoni** [comment]
check the stat menu under the @ symbol and post that. your calories stored say otherwise, and BMI of 19 isn't good irl, (19BMI at 5ft means your 90lbs).

BMI (Body Mass Index) for an 18yo healthy range is usually around 22BMI to 27BMI

**Comment by Alex-Folts** [comment]
Here something even more weird, stats of character i played year ago, very old save
![Scr1139](https://user-images.githubusercontent.com/5638906/87704519-315e0d00-c7a5-11ea-878d-889cdca83341.jpg)

**Comment by ZaricaDemoni** [comment]
that was back before 0.E Ellison

**Comment by Alex-Folts** [comment]
> check the stat menu under the @ symbol and post that.

![Scr1140](https://user-images.githubusercontent.com/5638906/87705187-340d3200-c7a6-11ea-89ca-e47988af43f5.jpg)

**Comment by ZaricaDemoni** [comment]
um, congratulations, I think you proved a theory on a bug, you should defiantly be considered underweight, yet just like me it says normal (when i was in this simular state), also wow you as tall as me irl. weather buddies!

**Comment by Alex-Folts** [comment]
I think it will change to underweight at 0:00 at night.

It was random character, and he was i think initially 193cm, i cut 10cm or so because was too weird.

**Comment by ZaricaDemoni** [comment]
19 BMI for that height is 140lbs not 90lbs as well.

**Comment by Alex-Folts** [comment]
> edit to why i mention parasite, your at 1950 with no exercise, unless that randomly generated which may be for all i know, im sitting at 1611 for no_exercise

I'm pretty sure this difference because of gender, there was some paper about 2000 and 2500 for average female and male calorie spending per day, so they seems take similar data and multiply by 0.8 and state that this is bare minimum for energy consumption per day. I'd argue this numbers because average person and person, who spent all day in bed doing absolutely nothing - difference between the two is not just 20% of energy consumption, some data required.

**Comment by ZaricaDemoni** [comment]
This was fixed by #42238 and is being closed as such.
